linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/java/org/apache/linkis/manager/rm/external/yarn/YarnResourceRequester.java

```diff
@@ -0,0 +1,426 @@
+        String id = app.get("id").asText();
+        String user = app.get("user").asText();
+        String applicationType = app.get("applicationType").asText();
+        Optional<YarnResource> yarnResource = getYarnResource(Optional.ofNullable(app), queueName);
+        if (yarnResource.isPresent()) {
+          YarnAppInfo appInfo =
+              new YarnAppInfo(id, user, state, applicationType, yarnResource.get());
+          appInfoList.add(appInfo);
+        }
+      }
+    }
+    return appInfoList;
+  }
+
+  @Override
+  public ResourceType getResourceType() {
+    return ResourceType.Yarn;
+  }
+
+  private JsonNode getResponseByUrl(String url, String rmWebAddress) {
+    HttpGet httpGet = new HttpGet(rmWebAddress + "/ws/v1/cluster/" + url);
+    httpGet.addHeader("Accept", "application/json");
+    Object authorEnable = this.provider.getConfigMap().get("authorEnable");
+    HttpResponse httpResponse = null;
+    if (authorEnable instanceof Boolean) {
+      if ((Boolean) authorEnable) {
+        httpGet.addHeader(HttpHeaders.AUTHORIZATION, "Basic " + getAuthorizationStr());
+      }
+    }
+    Object kerberosEnable = this.provider.getConfigMap().get("kerberosEnable");
+    if (kerberosEnable instanceof Boolean) {
+      if ((Boolean) kerberosEnable) {
+        String principalName = (String) this.provider.getConfigMap().get("principalName");
+        String keytabPath = (String) this.provider.getConfigMap().get("keytabPath");
+        String krb5Path = (String) this.provider.getConfigMap().get("krb5Path");
+        RequestKerberosUrlUtils requestKuu =
+            new RequestKerberosUrlUtils(principalName, keytabPath, krb5Path, false);
+        HttpResponse response =
+            requestKuu.callRestUrl(rmWebAddress + "/ws/v1/cluster/" + url, principalName);
+        httpResponse = response;
+      } else {
+        HttpResponse response = null;
+        try {
+          CloseableHttpClient httpClient = HttpClients.createDefault();
+          response = httpClient.execute(httpGet);
+        } catch (IOException e) {
+          logger.warn("getResponseByUrl failed", e);
+          throw new RuntimeException(e);
+        }
+        httpResponse = response;
+      }
+    } else {
+      HttpResponse response = null;
+      try {
+        CloseableHttpClient httpClient = HttpClients.createDefault();
+        response = httpClient.execute(httpGet);
+      } catch (IOException e) {
+        logger.warn("getResponseByUrl failed", e);
+        throw new RuntimeException(e);
+      }
+      httpResponse = response;
+    }
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    String entityString = "";
+    try {
+      entityString = EntityUtils.toString(httpResponse.getEntity());
+    } catch (IOException e) {
+      logger.warn("getResponseByUrl failed", e);
+      throw new RuntimeException(e);
+    }
+    JsonNode jsonNode = null;
+    try {
+      jsonNode = objectMapper.readTree(entityString);
```

Comment:
did you test with three level child queue , not sure if it works on failscheduler

Reply:
I haven't tested three level child queue yet, I would tested in next week and reply the result ,thanks for your time.

Reply:
Already tested with  three level child queue with following unit test and fairscheduler config
unit test:
![image](https://user-images.githubusercontent.com/16526279/236722882-413cdb42-1b88-4d46-a0eb-750a082981e8.png)

Reply:
you can try with multi child queue, for examples, you can define root.production.productionchild1,root.production.productionchild2,root.production.productionchild3, and test whether all this child queues parsed correctly.